test/connection_switching_test.rb

```diff
@@ -519,11 +519,20 @@ class ShardedModel < ActiveRecord::Base
         end
 
         it "will :include things via has_and_belongs associations correctly" do
+          if ActiveRecord::VERSION::MAJOR >= 4 && ActiveRecord::VERSION::MINOR >= 2
+            skip("There's a bug with Rails 4.2")
```

Comment:
It seems that your patch works with AR 4.2 as well – you can remove this `skip`.